src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -4467,6 +4467,9 @@ def updatePageWithParameters(self, line_dict, warn_user=True):
             if self.kernel_module.is_multiplicity_model:
                 self.kernel_module.multiplicity=multip
                 self.updateMultiplicityCombo(multip)
+        else:
+            self._n_shells_row = -1
+            self._num_shell_params = -1
```

Comment:
This is good, but maybe setting the default values for these two in lines 266/267 is even better?
(Seems to work as well)

Reply:
I think I found a conflict doing what you are suggesting during testing, but I can't remember what that was right now. I'll retest and add some inline documentation on why I chose this solution.